osquery/tables/system/darwin/process_open_descriptors.cpp

```diff
@@ -108,7 +108,8 @@ void genSocketDescriptor(int pid, int descriptor, QueryData &results) {
     Row r;
 
     r["pid"] = INTEGER(pid);
-    r["socket"] = INTEGER(descriptor);
+    r["fd"] = BIGINT(descriptor);
+    r["socket"] = BIGINT(si.psi.soi_so);
```

Comment:
This may be an API break, previously the "socket" include the descriptor number, this is not correct and on Linux was never the case (Linux used an inode-repr).